### ISSUE TITLE
Ability to configure DEFAULT_HEADERS

### ISSUE THREAD

**Issue by OR13** [digitalbazaar/edv-client#33]
In cases where http middleware is securing the edv, it is often necessary to set default headers for all requests. For example, using kong api gateway or cloudflare access control, you need to ensure that custom headers are present to get to the actual edv server.

**Comment by aljones15** [comment]
Good idea, but I do think we need to ensure it retains:

```
const DEFAULT_HEADERS = {Accept: 'application/ld+json, application/json'};
```

anyways this can done really easily by including it in the constructor of `EdvClient`.

Are there any objections? @dlongley @davidlehn @gannan08 @mattcollier

**Comment by OR13** [comment]
Yes, I'm happy to take a crack at it if you are really busy and its acceptable.

I would move defaultHeaders to the constructor as you mention... something like:

```
this.defaultHeaders = {
...defaultHeaders,
... DEFAULT_HEADERS
}
```

**Comment by aljones15** [comment]
> Yes, I'm happy to take a crack at it if you are really busy and its acceptable.
> 
> I would move defaultHeaders to the constructor as you mention... something like:
> 
> ```
> this.defaultHeaders = {
> ...defaultHeaders,
> ...DEFAULT_HEADERS
> }
> ```

I have no issue with you adding the ability to add additional headers, perhaps even at the request level if you request is going through something like an oauth endpoint that that might need it's Auth header refreshed every once in awhile, however I'm not an authority and this isn't my library. I would wait until a few people give the ok before spending time on it.

**Comment by dlongley** [comment]
I think making the `defaultHeaders` a member and enabling custom ones to be passed in the constructor is fine (and ensuring that the required default headers are always overlayed).

**Comment by OR13** [comment]
We should close this when the next version is cut and released.

**Comment by mattcollier** [comment]
Landed in v2.3, thanks @OR13
